src/Angor/Shared/WalletOperations.cs

```diff
@@ -127,40 +154,91 @@ public TransactionInfo AddInputsAndSignTransaction(string changeAddress, Transac
 
         if (coins.coins == null)
             throw new ApplicationException("No coins found");
+       
+        // did we spend all the coins?
+        var spendAll = coins.coins.Sum(s => s.Amount.Satoshi) == transaction.Outputs.Sum(o => o.Value.Satoshi);
 
-        var builder = new TransactionBuilder(network)
-            .AddCoins(coins.coins)
-            .AddKeys(coins.keys.ToArray())
-            .SetChange(BitcoinAddress.Create(changeAddress, network))
-            .ContinueToBuild(transaction)
-            .SendEstimatedFees(new FeeRate(Money.Satoshis(feeRate)))
-            .CoverTheRest();
+        if (spendAll)
```

Comment:
Didn't we agree to combine the 2 flows? or was it because of the change to PSBT

Reply:
that will anyway go away once we go full psbt so we agreed to keep it this way